apps/meteor/tests/e2e/omnichannel/omnichannel-livechat-api.spec.ts

```diff
@@ -345,12 +348,56 @@ test.describe('OC - Livechat API', () => {
 
 			const depId = departmentB._id;
 
-			await test.step('Expect setDepartment to change a guest department', async () => {
+			await test.step('Expect chat not be transferred', async () => {
 				await poLiveChat.page.evaluate((depId) => window.RocketChat.livechat.setDepartment(depId), depId);
+
+				await poAuxContext2.page.locator('role=navigation >> role=button[name=Search]').click();
+				await poAuxContext2.page.locator('role=search >> role=searchbox').fill(registerGuestVisitor.name);
+				await expect(
+					poAuxContext2.page.locator(`role=search >> role=listbox >> role=link >> text="${registerGuestVisitor.name}"`),
+				).not.toBeVisible();
+			});
+
+			await test.step('Expect registered guest to still be in dep1', async () => {
+				await poAuxContext.poHomeOmnichannel.sidenav.openChat(registerGuestVisitor.name);
 			});
+		});
+
+		test('OC - Livechat API - setDepartment - Default department', async () => {
+			const departmentB = departments[1].data;
```

Comment:
Any chance for this be undefined?

Reply:
If any of the departments is undefined, the test should fail since something went wrong with the department creation